protocol/messenger.go

```diff
@@ -3525,6 +3528,13 @@ func (m *Messenger) saveDataAndPrepareResponse(messageState *ReceivedMessageStat
 		if ok {
 			contactsToSave = append(contactsToSave, contact)
 			messageState.Response.AddContact(contact)
+
+			_, ok := m.allContacts.Load(id)
+			if !ok {
+				// If the contact is not in the allContacts map, it means it's a new contact
+				// and we need to add it to the allContacts map.
+				m.allContacts.Store(id, contact)
+			}
```

Comment:
Isn't it done earlier during message processing? Or did you face some bug? 🤔

Reply:
Yeah, without this, my test doesn't pass. 
On this line:
```go
s.Require().Len(bob2.Contacts(), 2)
```

So either I'm not calling the right function(s) to parse the contacts or I found a small bug?

Reply:
weird. But I guess your fix is correct 👍